### Supondo um filme com as seguintes características:

{'Series_Title': 'The Shawshank Redemption',
'Released_Year': '1994',
'Certificate': 'A',
'Runtime': '142 min',
'Genre': 'Drama',
'Overview': 'Two imprisoned men bond over a number of years,
finding solace and eventual redemption through acts of common
decency.',
'Meta_score': 80.0,
'Director': 'Frank Darabont',
'Star1': 'Tim Robbins',
'Star2': 'Morgan Freeman',
'Star3': 'Bob Gunton',
'Star4': 'William Sadler',
'No_of_Votes': 2343110,
'Gross': '28,341,469'}
<br>
#### Calcular IMDB

In [4]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
from sklearn.pipeline import Pipeline
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error
import joblib
import pandas as pd 
import numpy as np

In [5]:
df = pd.read_csv("../data/cinema_processed.csv")

In [6]:
features = ["Meta_score", "Runtime", "No_of_Votes", "Gross", "Genre"] 
X = df[features].copy() 
y = df["IMDB_Rating"] 

categorico = ["Genre"] 
numerico = ["Meta_score", "Runtime", "No_of_Votes", "Gross"] 

preprocessamento = ColumnTransformer( 
    transformers=[
        ("cat", OneHotEncoder(handle_unknown="ignore"), categorico),
        ("num", "passthrough", numerico) 
        ]
    ) 

model = Pipeline(steps=[ ("preprocessamento", preprocessamento), ("regressor", RandomForestRegressor(n_estimators=300, random_state=42)) ]) 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42) 


model.fit(X_train, y_train)
joblib.dump(model, '../src/model/modelo_imdb.pkl')

y_pred = model.predict(X_test)
rmse = np.sqrt(mean_squared_error(y_test, y_pred))
print(f"RMSE: {rmse:.2f}")

filme = pd.DataFrame([{ "Meta_score": 80.0, "Runtime": 142, "No_of_Votes": 2343110, "Gross": 28341469, "Genre": "Drama" }])

predicao_imdb = model.predict(filme) 
print(f"Nota IMDB prevista: {predicao_imdb[0]:.2f}")




RMSE: 0.22
Nota IMDB prevista: 8.76


#### Resultado

RMSE = “quão confiáveis são minhas previsões em média” → 0.22 indica boa acuracia. <br>
Nota prevista = “o que meu modelo acha que esse filme merece no IMDB” → 8.76 mostra que o modelo aprendeu padrões gerais, mas não captura perfeitamente a nota exata.